src/usecases/retrieveAverageVisitsPerDay.js

```diff
@@ -0,0 +1,24 @@
+const retrieveAverageVisitsPerDay = ({ getDb }) => async (trustId) => {
+  if (!trustId) return { error: "A trustId must be provided." };
+
+  const db = await getDb();
+  const [{ average_visits_per_day: averageVisitsPerDay }] = await db.any(
+    `SELECT to_char(COALESCE(AVG(visits), 0), '999D9') as average_visits_per_day FROM(
+      SELECT COUNT (distinct events.visit_id ) AS visits
+      FROM events
+      LEFT JOIN scheduled_calls_table ON events.visit_id = scheduled_calls_table.id
+      LEFT JOIN wards ON scheduled_calls_table.ward_id = wards.id
+      LEFT JOIN trusts ON wards.trust_id = trusts.id
+      WHERE trusts.id = $1 AND events.action='join-visit'
+      GROUP BY date(scheduled_calls_table.call_time)
```

Comment:
Should we be grouping by the call_time or the time of the event ?

Reply:
Grouping by `call_time`. Event time doesn't work.

Reply:
Could we clarify why it doesn't work ? Is it not suitable for this metric?

Reply:
I've not got to the bottom of why, but for some reason the values come back all wrong.

Reply:
Were we using MockDate to set the correct date before capturing events ? (like https://github.com/madetech/nhs-virtual-visit/blob/master/src/usecases/retrieveAverageVisitTimeByTrustId.contractTest.js#L40-L54)

Reply:
So I took a look at this and the test pass with the mock date (woo). However running the query against the staging data set still gives wildly different results. Not sure what route to do down with this as it needs investigation to determine which is best, but I'm not sure it's worth the time. What were you expecting to gain from using event time vs call time?

Reply:
Just wanted to make sure we were using the right timestamp for the metric (as we use the event time for other metrics). But since this one is for daily visits it makes sense to use the call_time as it's unlikely we'll have people joining days after the call is scheduled (since the data is wiped)